client/app/components/app-view/index.js

```diff
@@ -0,0 +1,64 @@
+import template from './template.html';
+
+export default function init(ngModule) {
+  ngModule.component('appView', {
+    template,
+    controller($rootScope, $route, Auth) {
+      this.showHeaderAndFooter = false;
+
+      $rootScope.$on('$routeChangeStart', (event, route) => {
+        if (route.$$route.authenticated) {
+          // For routes that need authentication, check if session is already
+          // loaded, and load it if not.
+          Auth.logger('Requested authenticated route: ', route);
+          if (Auth.isAuthenticated()) {
+            this.showHeaderAndFooter = true;
+          } else {
+            event.preventDefault();
+            Auth.loadSession().then(() => {
+              if (Auth.isAuthenticated()) {
+                this.showHeaderAndFooter = true;
+                Auth.logger('Loaded session');
+                $route.reload();
+              } else {
+                throw new Error('Need to login');
```

Comment:
I'm assuming this will be caught by `catch` below, why not call `Auth.login` here?